### DISCUSSION TITLE
Authorize location on webview

### DISCUSSION THREAD

**Discussion by afmaster** [beeware/toga#1690] [Q&A]
How to allow an webview app, using Beeware/toga, to show permission request popup? 

The same page works fine by accessing it via mobile chrome, but do not work in webview because the popup request to allow location tracking does not appear. Is there a way to allow tracking location?

**Comment by freakboy3742** [comment]
I can't say I've tried doing this, so I unfortunately can't provide a worked solution.

I'm guessing what will be happening behind the scenes is either:
1. The web view is triggering a callback to ask the user for permissions, and the default system handler for that permission is rejecting permission. 
2. The web view needs to be created with a specific permission turned on. 
3. The app needs to be created requesting location tracking permissions.
 
If it's 1 or 2, the fix will require a platform-specific piece of code in the Toga backend creating the webview.

If it's 3, it falls into a known limitation of app packaging with Briefcase - we don't currently allow customisation of permissions. This is something we intend to look at in the near future; in the meantime, you need to manually update the platform-specific declarations in your app bundle. However, I can't advise what those will be, as you haven't told us which platform you're on.

**Comment by afmaster** [comment]
I'm using Android. But it seems not work in ios too. 

I think there is some config tricks in Android manifest. I've tried some suggestions made[ here.](https://stackoverflow.com/questions/5329662/android-webview-geolocation).

None of them worked for me. But I have to admit that I am not expert in android development and native packages....

**Reply by mhsmith** [reply]
If you post the details of exactly what you tried and what happened, then we may be able to help you.
